academic_observatory_workflows/database/sql/create_doi.sql.jinja2

```diff
@@ -30,37 +32,44 @@ FROM
   WHERE ARRAY_LENGTH(issued.date_parts) > 0)
 ),
 
+{# this query builds the .affiliation section of the final doi table. The primiary purpose of this is to allow the aggregrate_doi query #}
 affiliations_temp_table as (
 SELECT
   extras.doi as doi,
 
   institutions,
+  {# The four affiliation types below (Country, SubRegion, Region and Grouping, take their values from the SubQuery Below 'LEFT JOIN( SELECT...' however they need to be GROUPed to ensure only unique lists. For example, two institutions in one country result in two records of the same counry and a squashed into a single instance #}
   ARRAY(SELECT as STRUCT identifier, MAX(name) as name, ["Country"] as types, CAST(ARRAY_AGG(home_repo IGNORE NULLS) AS ARRAY<STRING>) as home_repo, CAST(MAX(country) as STRING) as country, CAST(MAX(country_code) as STRING) as country_code, CAST(MAX(country_code_2) as STRING) as country_code_2, CAST(MAX(region) as STRING) as region, CAST(MAX(subregion) as STRING) as subregion, CAST(NULL as STRING) as coordinates, COUNT(*) as count, ARRAY_AGG(DISTINCT member_identifier IGNORE NULLS) as members FROM UNNEST(countries) GROUP BY identifier) as countries,
   ARRAY(SELECT as STRUCT identifier, MAX(name) as name, ["Subregion"] as types, CAST(ARRAY_AGG(home_repo IGNORE NULLS) AS ARRAY<STRING>) as home_repo, CAST(MAX(country) as STRING) as country, CAST(MAX(country_code) as STRING) as country_code, CAST(MAX(country_code_2) as STRING) as country_code_2, CAST(MAX(region) as STRING) as region, CAST(MAX(subregion) as STRING) as subregion, CAST(NULL as STRING) as coordinates, COUNT(*) as count, ARRAY_AGG(DISTINCT member_identifier IGNORE NULLS) as members  FROM UNNEST(subregions) GROUP BY identifier) as subregions,
   ARRAY(SELECT as STRUCT identifier, MAX(name) as name, ["Region"] as types, CAST(ARRAY_AGG(home_repo IGNORE NULLS) AS ARRAY<STRING>) as home_repo, CAST(MAX(country) as STRING) as country, CAST(MAX(country_code) as STRING) as country_code, CAST(MAX(country_code_2) as STRING) as country_code_2, CAST(MAX(region) as STRING) as region, CAST(MAX(subregion) as STRING) as subregion, CAST(NULL as STRING) as coordinates, COUNT(*) as count, ARRAY_AGG(DISTINCT member_identifier IGNORE NULLS) as members FROM UNNEST(regions) GROUP BY identifier) as regions,
   ARRAY( SELECT as STRUCT grouping_entity.group_id as identifier, MAX(grouping_entity.group_name) as name, ["Grouping"] as types, CAST([] AS ARRAY<STRING>) as home_repo, CAST(NULL as STRING) as country, MAX(grouping_entity.country_code) as country_code, MAX(grouping_entity.country_code) as country_code_2, CAST(NULL as STRING) as region, CAST(NULL as STRING) as subregion, CAST(NULL as STRING) as coordinates, ARRAY_AGG(DISTINCT grid_group.member_identifier IGNORE NULLS) as members FROM UNNEST(grid_groups) as grid_group, UNNEST(grid_group.groupings) as grouping_entity GROUP BY grouping_entity.group_id) as groupings,
```

Comment:
Perhaps the comments and the SQL could be broken into multiple lines to make it easier to read., e.g. keeping the lines under 120 characters long.

Reply:
not perfectly 120 characters. But I went through and broke up many of the longer lines, and ensured the comments were more consistently styled